node/src/accountant/scanners/mod.rs

```diff
@@ -938,13 +939,20 @@ impl ReceivableScanner {
                     ),
                 }
                 match txn.commit() {
-                    Ok(_) => {
-                        debug!(logger, "Updated start block to: {}", new_start_block)
-                    }
+                    Ok(_) => debug!(logger, "Updated start block to: {}", new_start_block),
                     Err(e) => panic!("Commit of received transactions failed: {:?}", e),
                 }
             } else {
-                warning!(logger, "The new_start_block ({}) is less than the current_start_block ({}). This is not a problem but by checking we avoid rescanning the same blocks again later.", &new_start_block, &current_start_block);
+                match txn.commit() {
+                    Ok(_) => debug!(logger, "The new_start_block ({}) is not greater than the current_start_block ({}). \
+                    Committed received transactions.", &new_start_block, &current_start_block),
+                    Err(e) => panic!("Commit of received transactions failed: {:?}", e),
```

Comment:
Another and even less cool thing I'm witnessing is that I can't see any new tests that support this new code. There should be at least one test modified for the happy path (as I expect there might've been some sort of happy path before too), or a fully-new test, if somebody did a floppy job. Plus, there definitely should be one brand new test that concentrates on the situation when the committing results in an error and so a panic. 

These things are a must.  

Please make this right. 
You'll need to use this special version of a database transaction: `TransactionSafeWrapper`. Make it by using `TransactionSafeWrapper::new_with_builder`. Either get some inspiration from tests that have it in use or just let the code navigate you through since the builder is designed to be the only way of how you can construct a mock version which you can control. And you do want to.

You need to assert by `commit_params` that this function was called, that for the happy path. In the sad path, you are just obligated to ensure the mock will return an error during that function call. So choose an error to be spit out, by setting it up with `commit_result`.    

You're still possessing quite a lot of work to do. Please keep yourself better-focused this time.

Reply:
I mentioned this in the call on Sunday. Why don't I already find `TransactionSafeWrapper` in use in `accountant/scanners/mod.rs` I didn't introduce `txn.commit()` for the first time.

I tried to trace even the happy path using the added logging but I didn't find it.

Reply:
If you haven't seen my [comment](https://github.com/MASQ-Project/Node/pull/374/commits/d93e62cc8ca4e8d381553e49c9e23816ad7fc174#r1776159174) just before this one, please read.

I will provide adequate coverage for these changes and hopefully eliminate the extra paths. so it will be as it was before GH-606 except for passing `Some(msg.new_start_block)` to be persisted.